### PR TITLE
Add a way to avoid core-dumps on death tests

### PR DESCRIPTION
**Pull request by mchacki**
This is test-only functionality without production implications.
Will however help with our Jenkins setup.
We have not used this mechanism in any released version, so no back port required.

Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6525/

### CODE REVIEW

File: tests/Mocks/Death_Test.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      [&]() {                                 \
+        rlimit core_limit;                    \
+        core_limit.rlim_cur = 0;              \
+        core_limit.rlim_max = 0;              \
+        setrlimit(RLIMIT_CORE, &core_limit);  \
+        func;                                 \
+      }(),                                    \
+      assertion)
+
+#endif
```
Comment (mchacki) [inline tests/Mocks/Death_Test.h]:
```suggestion
#endif

```

File: tests/Mocks/Death_Test.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/// invalid in production code and not accidentially removed on refactorings.
+/// However every such death test will generate a core-dump, even if the test is successful
+/// This is inconvenient as it unnecessarily bloats up HDD usage and hides releveant coredumps
+/// So this thin macro wraps around the GTEST :: EXPECT_DEATH macro and disables coredumps
+/// only within the expected forked process
+
+#ifndef ARANGODB_TESTS_MOCKS_DEATH_TEST_CHANGER_H
+#define ARANGODB_TESTS_MOCKS_DEATH_TEST_CHANGER_H 1
+
+#define CORE_FREE_DEATH_TEST(func, assertion) \
```
Comment (markuspf) [inline tests/Mocks/Death_Test.h]:
```suggestion
#define EXPECT_DEATH_CORE_FREE(func, assertion) \
```

File: tests/Aql/SubqueryEndExecutorTest.cpp [hunk 3]
```diff
@@ -282,7 +283,8 @@ TEST_F(SubqueryEndExecutorTest_DeathTest, no_shadow_row) {
   outputBlock.reset(new AqlItemBlock(itemBlockManager, inputBlock->size(), 1));
   OutputAqlItemRow output{std::move(outputBlock), _infos.getOutputRegisters(),
                           _infos.registersToKeep(), _infos.registersToClear()};
-  EXPECT_DEATH(std::tie(state, std::ignore) = testee.produceRows(output), ".*");
+  CORE_FREE_DEATH_TEST(std::tie(state, std::ignore) = testee.produceRows(output),
```
Comment (markuspf) [inline tests/Aql/SubqueryEndExecutorTest.cpp]:
```suggestion
  EXPECT_DEATH_CORE_FREE(std::tie(state, std::ignore) = testee.produceRows(output),
```

File: tests/Aql/SubqueryEndExecutorTest.cpp [hunk 4]
```diff
@@ -299,5 +301,6 @@ TEST_F(SubqueryEndExecutorTest_DeathTest, misplaced_irrelevant_shadowrow) {
...
```
Comment (markuspf) [inline tests/Aql/SubqueryEndExecutorTest.cpp]:
```suggestion
  EXPECT_DEATH_CORE_FREE(std::tie(state, std::ignore) = testee.produceRows(output),
```

**Comment by mchacki** [comment]
I'll go for your naming. Thanks!

**Comment by mchacki** [comment]
Newest Jenkins after long weekend:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6604/

**Pull request merged**